Source/WTF/wtf/glib/SocketConnection.cpp

```diff
@@ -109,44 +109,36 @@ static inline bool messageIsByteSwapped(MessageFlags flags)
 #endif
 }
 
-IGNORE_CLANG_WARNINGS_BEGIN("unsafe-buffer-usage-in-libc-call")
 bool SocketConnection::readMessage()
 {
     if (m_readBuffer.size() < sizeof(uint32_t))
         return false;
 
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GLib port.
-    auto* messageData = m_readBuffer.mutableSpan().data();
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
-    uint32_t bodySizeHeader;
-    memcpy(&bodySizeHeader, messageData, sizeof(uint32_t));
-    messageData += sizeof(uint32_t);
-    bodySizeHeader = ntohl(bodySizeHeader);
-    Checked<size_t> bodySize = bodySizeHeader;
-    MessageFlags flags;
-    memcpy(&flags, messageData, sizeof(MessageFlags));
-    messageData += sizeof(MessageFlags);
-    auto messageSize = sizeof(uint32_t) + sizeof(MessageFlags) + bodySize;
+    auto messageData = m_readBuffer.span();
+    const size_t bodySize = ntohl(consumeAndReinterpretCastTo<uint32_t>(messageData));
+    const size_t messageSize = sizeof(uint32_t) + sizeof(MessageFlags) + bodySize;
```

Comment:
I would use `Checked<size_t>` here unless you've squinted very carefully at the rest of this function. It's clear enough that these two expressions will never overflow, but this has impacts on the rest of this function as well. E.g. `m_readBuffer.size() - messageSize.value()` was previously checked, but no longer is.